src/hotspot/share/ci/ciFlags.hpp

```diff
@@ -39,37 +38,37 @@ class ciFlags {
+    _flags(flags), _stable(is_stable), _initialized_final_update(is_initialized_final_update) { }
 
 public:
   // Java access flags
-  bool is_public               () const { return (_flags & JVM_ACC_PUBLIC                    ) != 0; }
-  bool is_private              () const { return (_flags & JVM_ACC_PRIVATE                   ) != 0; }
-  bool is_protected            () const { return (_flags & JVM_ACC_PROTECTED                 ) != 0; }
-  bool is_static               () const { return (_flags & JVM_ACC_STATIC                    ) != 0; }
-  bool is_final                () const { return (_flags & JVM_ACC_FINAL                     ) != 0; }
-  bool is_synchronized         () const { return (_flags & JVM_ACC_SYNCHRONIZED              ) != 0; }
-  bool is_super                () const { return (_flags & JVM_ACC_SUPER                     ) != 0; }
-  bool is_volatile             () const { return (_flags & JVM_ACC_VOLATILE                  ) != 0; }
-  bool is_transient            () const { return (_flags & JVM_ACC_TRANSIENT                 ) != 0; }
-  bool is_native               () const { return (_flags & JVM_ACC_NATIVE                    ) != 0; }
-  bool is_interface            () const { return (_flags & JVM_ACC_INTERFACE                 ) != 0; }
-  bool is_abstract             () const { return (_flags & JVM_ACC_ABSTRACT                  ) != 0; }
+  bool is_public               () const { return _flags.is_public();       }
+  bool is_private              () const { return _flags.is_private();      }
+  bool is_protected            () const { return _flags.is_protected();    }
+  bool is_static               () const { return _flags.is_static();       }
+  bool is_final                () const { return _flags.is_final();        }
+  bool is_synchronized         () const { return _flags.is_synchronized(); }
+  bool is_super                () const { return _flags.is_super();        }
+  bool is_volatile             () const { return _flags.is_volatile();     }
+  bool is_transient            () const { return _flags.is_transient();    }
+  bool is_native               () const { return _flags.is_native();       }
+  bool is_interface            () const { return _flags.is_interface();    }
+  bool is_abstract             () const { return _flags.is_abstract();     }
   bool is_stable               () const { return _stable; }
   // In case the current object represents a field, return true if
   // the field is modified outside of instance initializer methods
   // (or class/initializer methods if the field is static) and false
   // otherwise.
-  bool has_initialized_final_update() const { return _intialized_final_update; };
+  bool has_initialized_final_update() const { return _initialized_final_update; };
 
   // Conversion
-  jint   as_int()                      { return _flags; }
+  jint   as_int()                      { return _flags.as_unsigned_short(); }
```

Comment:
It is unclear to me whether the fact we are dealing with u2 should be exposed in this API as well.

Reply:
I don't think it should be.